app/controllers/bz_controller.rb

```diff
@@ -194,137 +138,8 @@ def grade_details
-          row << "#{criterion["description"]} -- #{assignment.name}"
-        end
-      end
-
-      csv << row
-
-
-      # data
-      students_done = {}
-      students.each do |student_obj|
-        next if students_done[student_obj.id]
-        students_done[student_obj.id] = true
-
-        row = []
-        # name
-        row << student_obj.id
-        row << student_obj.name
-        row << student_obj.email
-
-        assignments_info.each do |assignment_info|
-          sg = assignment_info[:sg]
-          assignment = assignment_info[:assignment]
-          rubric = assignment_info[:rubric]
-          criteria = assignment_info[:criteria]
-          sections = assignment_info[:sections]
-          sections_points_available = assignment_info[:sections_points_available]
-
-
-          student = nil
-
-          sg["context"]["students"].each do |student_sg|
-            if student_sg["id"].to_i == student_obj.id.to_i
-              student = student_sg
-              break
-            end
-          end
-          if student.nil?
-            student = {}
-            student["name"] = student_obj.name
-            student["rubric_assessments"] = []
-          end
-
-          latest_assessment = student["rubric_assessments"].last
-
-          # total score
-          if latest_assessment
-            row << "#{(latest_assessment["score"].to_f * 100 / assignment.points_possible.to_f).round(2)}%"
-          else
-            row << "0"
-          end
-
-          # section averages...
-
-          section_scores = []
-
-          criteria.each do |criterion|
-            points = 0.0
-            if latest_assessment
-              latest_assessment["data"].each do |datum|
-                points = datum["points"].to_f if datum["criterion_id"] == criterion["criterion_id"]
-              end
-            end
-            if section_scores[criterion["section"].to_i].nil?
-              section_scores[criterion["section"].to_i] = 0.0
-            end
-            section_scores[criterion["section"].to_i] += points # zero-based array of one-based sections
-          end
-
-
-          sections.each do |idx|
-            ss = section_scores[idx.to_i]
-            if sections_points_available[idx.to_i].nil?
-              row << ss
-            else
-              row << "#{(ss * 100 / sections_points_available[idx.to_i]).round(2)}%"
-            end
-          end
-
-          # individual breakdown...
-
-          criteria.each do |criterion|
-            points = 0.0
-            if latest_assessment
-              latest_assessment["data"].each do |datum|
-                points = datum["points"].to_f if datum["criterion_id"] == criterion["criterion_id"]
-              end
-            end
-            row << points
-          end
-        end
-        csv << row
-      end
-    end
-
-    respond_to do |format|
-      format.csv { render text: csv }
-    end
+    download = BzController::GradesDownload.new(current_user.email, params)
```

Comment:
where is this class defined?

Reply:
It's defined in the lib directory, where all ruby-obedient custom libraries live :)

Reply:
I didn't like to do this because it breaks convention, but I followed the existing formula of including it further down in the same file. It really should be in its own file, but I followed the precedent of the ExportWork class in the same file.

Reply:
There's a `Export::GradeDownload` but no `BzController::GradesDownload`, and the `ExportGrades` class doesn't appear to be used. (wouldn't it be great if we were just using a real programming language with a compiler that could call this out?)

Reply:
You're right, my mistake. And no compiler needed, if we had tests :)